refactor(checkbox-component): example checkbox component with variant red

navin-moorthy opened the pull request:
Starter code to begin with Checkbox Component.

We can evolve from this based on our discussion.

navin-moorthy committed:
refactor(checkbox-component): example checkbox component with variant red

prasanna1211 commented:
but yes don't spread the props may be ? handle props one by one.
If there are style props that are going to be common, keep it in a separate file and share it across.

navin-moorthy commented:
> but yes don't spread the props may be ? handle props one by one.
> If there are style props that are going to be common, keep it in a separate file and share it across.

Sure, but in this case we are not providing any props so the prop spreading can be removed. But show us an example on how to handle prop.

govindsingh55 commented:
Only the default props that we want to define or different props than chakra needs ahould have condition.

navin-moorthy commented:
> Only the default props that we want to define or different props than chakra needs have condition.

Can you elaborate please? I don't get it.

prasanna1211 commented:
Keep proceeding I will join once everything is completed from your side.

navin-moorthy committed:
refactor(checkbox component): remove the prop spread

SamrithaS committed:
feat: checkbox color change

navin-moorthy commented on package.json:
```diff
@@ -95,7 +95,8 @@
     "sanitize.css": "8.0.0",
-    "styled-components": "4.2.0"
+    "styled-components": "^4.2.0",
+    "styled-system": "^5.1.5"
```
Remove these dependencies and `npm install` to update your node_modules.
Not so important, but a good practise

navin-moorthy commented on app/theme.js:
```diff
@@ -0,0 +1,9 @@
+  ...theme,
+  colors: {
+    ...theme.colors,
+    brand: '#5D5E5E',
```
It seems we are getting a warning as below to add variant
![image](https://user-images.githubusercontent.com/39694575/77643750-eca82f00-6f85-11ea-88be-f1529f86edb3.png)
change this line with the generated colors will solve the issue - https://coolors.co/
```json
brand: {
      '50': '#8ED0E9',
      '100': '#7BC8E6',
      '200': '#69C0E2',
      '300': '#56B8DF',
      '400': '#43B0DB',
      '500': '#31A9D8',
      '600': '#2D9AC5',
      '700': '#298BB1',
      '800': '#247B9E',
      '900': '#206C8A',
    },
```

SamrithaS replied:
I'll do it

SamrithaS committed:
fix: fixed the warning - checkbox color

navin-moorthy commented:
Last commit, looks fine, able to run and get the expected output
![image](https://user-images.githubusercontent.com/39694575/77647682-db165580-6f8c-11ea-8458-f2a933fa00ec.png)

navin-moorthy committed:
refactor(index.html): add inter font family

Add Inter Font Family from https://rsms.me/inter/inter.css via html link

navin-moorthy committed:
feat(theme.js): add Inter styles to theme.js for body and heading

Add Inter font family to theme.js body and heading to use in our ecosystem

[RFC] Checkbox Component #16

navin-moorthy committed:
feat(app.js): add CSSReset to normalize our browser CSS

navin-moorthy committed:
feat(checkbox index.js): add Control Box for Checkbox of Chakra to create custom checkbox

[RFC] Checkbox Component #16

navin-moorthy committed:
refactor(homepage index.js): add more local examples

navin-moorthy commented:
Added custom font styles

navin-moorthy committed:
chore(package.json): remove un-necessary libraries added

SamrithaS committed:
feat: added border color

navin-moorthy committed:
fix(package.json): add styled-components dependency

SamrithaS committed:
feat: added focus and border radius for checkbox

SamrithaS committed:
fix: fixed margin left

navin-moorthy committed:
style(checkbox): add hover style

navin-moorthy committed:
fix(checkbox): correct inset shadow

navin-moorthy committed:
fix(index.html): change the Inter Font Family source

govindsingh55 committed:
refactor(checkbox icon changes): icon changes

navin-moorthy commented on app/theme.js:
```diff
@@ -25,4 +26,21 @@ export default {
+            fill="#fafafa"
+            d="M385.75 171.586c8.34 8.34 8.34 21.82 0 30.164L247.082 340.414c-4.16 4.16-9.621 6.254-15.082 6.254s-10.922-2.094-15.082-6.254l-69.332-69.332c-8.344-8.34-8.344-21.824 0-30.164 8.34-8.344 21.82-8.344 30.164 0l54.25 54.25 123.586-123.582c8.34-8.344 21.82-8.344 30.164 0zm0 0"
+          />
+        </svg>
```
Icon should be pasted only after minimizing with [SVGO](https://jakearchibald.github.io/svgomg/)
Also we only need the check icon not the whole box
```html
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 10 7"><path fill="none" fill-rule="evenodd" stroke="#FFF" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M1.5 3.788L3.651 6l5.058-5"/></svg>
```

as below

```json
icons: {
    ...theme.icons,
    customCheck: {
      path: (
        <path
          fill="none"
          fillRule="evenodd"
          stroke="#FFF"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
          d="M1.5 3.788L3.651 6l5.058-5"
        />
      ),
      viewBox: '0 0 10 7',
    },
  },
```

govindsingh55 replied:
it's already minified SVGOMG

navin-moorthy replied:
We only need the check svg not the box as well.
See my path, how simple it and how I have declared the viewbox.
That will give the checkbox correct style.

navin-moorthy replied:
![image](https://user-images.githubusercontent.com/39694575/77744018-f772ca80-703e-11ea-835b-129e6447e49b.png)

navin-moorthy commented on app/components/Checkbox/index.js:
```diff
@@ -34,14 +34,22 @@ const Checkbox = ({ children }) => (
+Checkbox.defaultProps = {
+  iconName: 'customCheck',
+  iconSize: '10px',
+  label: 'default label',
```
Why are we exposing the proptypes for these base styles. We are not going to expose these props,
This is why before making a commit, we all need to agree on making a change to our codebase.
We need the children to be exposed outside,
since our checkbox is used like this `<Checkbox>Three</Checkbox>`

govindsingh55 replied:
checkbox label is gonna be string, that's why make the children to label.
i have exposed some props like iconName, and iconSize that can be modified from outside.

navin-moorthy replied:
Kindly refer how Chakra has built the Checkbox label, it should be only children
https://github.com/chakra-ui/chakra-ui/blob/master/packages/chakra-ui/src/Checkbox/index.js

Icon Name and Icon Size is going to be the same, if we are going to expose that prop that should have gone through the discussion in the RFC.

We have already started the conversion, we wont be exposing icon name, because throughout the project, the same name will be used.

We can expose the iconsize but you can mention that in the RFC.

navin-moorthy committed:
feat(theme.js): add custom check icon

navin-moorthy committed:
feat(checkbox): add the customCheck to our checkbox

SamrithaS committed:
feat: disabled prop added

navin-moorthy commented on app/theme.js:
```diff
@@ -19,6 +19,12 @@ export default {
+    },
+    disabled: {
+      borderColor: '#E7E8E9',
+      color: '#CACBCC',
```
Change this into same object like 

```suggestion
    checkbox: {
      borderColor: '#8C8D8E',
      color: '#333536',
      shadow: '#31a7d840',
    disabled: {
      borderColor: '#E7E8E9',
      color: '#CACBCC',
    },
```

because this is checkbox disabled.

navin-moorthy replied:
Change the above object and change accordingly in checkbox file

SamrithaS replied:
okay

SamrithaS committed:
fix: checkbox disabled styles fix

navin-moorthy committed:
feat(checkbox): add onBlur,onChange, onFocus & isChecked

navin-moorthy committed:
chore(example): add more examples for each prop exposed

SamrithaS committed:
feat: defaultIsChecked prop added

navin-moorthy committed:
fix(example): fix bug with the example for isChecked with onChange

navin-moorthy committed:
chore(example): update example with useState for isChecked

navin-moorthy committed:
feat(checkbox): add id, name, value prop

navin-moorthy committed:
fix(checkbox): fix the isDisabled cursor blackhole bug

navin-moorthy committed:
feat(storybook): add storybook to the project with default demos

prasanna1211 committed:
feat: add allowable props

Allowed props for every chakra component

prasanna1211 committed:
feat: overwrides for checkbox

BREAKING CHANGE: Can add in structured styles

SamrithaS committed:
feat: package.json repo, readme installation and usage

prasanna1211 committed:
feat(override changes): override changes for checkbox

changes in override for checkbox

navin-moorthy committed:
feat(storybook): add storybook for checkbox component

govindsingh55 committed:
refactor(default style): added default style to checkbox

n

govindsingh55 committed:
refactor(input isdisabled props change): isDisabled props change for InVisible component

govindsingh55 committed:
Revert "Checkbox default props"

govindsingh55 committed:
refactor(getallowedprops changes): allowing partial props pass with styledProps

navin-moorthy committed:
refactor(checkbox): improve theme.js and allow override styles (#24)

* refactor(theme.js): improve styling so that all the styles come from theme.js

* refactor(theme): handle width and other props with more verbose version

navin-moorthy commented:
Review this PR #17 and merge it @prasanna1211

prasanna1211 committed:
added docs

prasanna1211 committed:
added docs
